src/storage/compression/float_compression.cpp

```diff
@@ -158,18 +164,18 @@ void FloatCompression<T>::setValuesFromUncompressed(const uint8_t* srcBuffer,
         integerEncodedValues[i] = encodedValue;
     }
 
-    getEncodedFloatBitpacker(metadata)->setValuesFromUncompressed(
+    getEncodedFloatBitpacker(metadata).setValuesFromUncompressed(
         reinterpret_cast<const uint8_t*>(integerEncodedValues.data()), 0, dstBuffer, dstOffset,
         numValues, metadata.getChild(BITPACKING_CHILD_IDX), nullMask);
 }
 
 template<std::floating_point T>
-std::unique_ptr<CompressionAlg> FloatCompression<T>::getEncodedFloatBitpacker(
+const CompressionAlg& FloatCompression<T>::getEncodedFloatBitpacker(
     const CompressionMetadata& metadata) const {
     if (metadata.getChild(BITPACKING_CHILD_IDX).isConstant()) {
-        return std::make_unique<ConstantCompression>(getBitpackingLogicalType<T>());
+        return constantEncodedFloatBitpacker;
```

Comment:
Is this still necessary with #4166? Constant compression should be handled higher up now. Or is there some special meaning to the float compression having a constant child compression type?

Reply:
Let me check on this

Reply:
I feel like the changes in #4166 only affect how the compression metadata is generated, I don't think we have a similar mechanism for handling constant compression higher up when picking a compression algorithm to use since we still have code like [this](https://github.com/kuzudb/kuzu/blob/master/src/storage/compression/compression.cpp#L739).